tools/getnodeversion.py

```diff
@@ -13,4 +19,41 @@
   if re.match('#define NODE_PATCH_VERSION', line):
     patch = line.split()[2]
 
-print '%(major)s.%(minor)s.%(patch)s'% locals()
+  if not no_tag:
+    tag_match = re.match('#define NODE_TAG (.*)', line)
+    if tag_match:
+      tag = tag_match.group(1)
+
+def get_version_string(major, minor, patch, tag, msi=False, no_tag=False):
+  version_string = '%(major)s.%(minor)s.%(patch)s'% locals()
+  if tag:
+    if msi:
+      rc_number_match = re.match('.*(\d+)$', tag)
+      if rc_number_match and rc_number_match.group(1):
+        rc_number = rc_number_match.group(1)
+        version_string += '.%(rc_number)s'% locals()
+      else:
+        sys.stderr.write('Could not determine proper rc number for msi ' +
+                         'version, exiting')
```

Comment:
At a glance this looks great. I didn't see this until now and an wiped out
from work. I'll look at this more in depth tomorrow.

On Fri, Apr 3, 2015, 15:33 orangemocha <notifications@github.com> wrote:

> In tools/getnodeversion.py
> <https://github.com/joyent/node/pull/14412#discussion_r27760454>:
>
> > +  if not no_tag:
> > +    tag_match = re.match('#define NODE_TAG (.*)', line)
> > +    if tag_match:
> > +      tag = tag_match.group(1)
> > +
> > +def get_version_string(major, minor, patch, tag, msi=False, no_tag=False):
> > +  version_string = '%(major)s.%(minor)s.%(patch)s'% locals()
> > +  if tag:
> > +    if msi:
> > +      rc_number_match = re.match('.*(\d+)$', tag)
> > +      if rc_number_match and rc_number_match.group(1):
> > +        rc_number = rc_number_match.group(1)
> > +        version_string += '.%(rc_number)s'% locals()
> > +      else:
> > +        sys.stderr.write('Could not determine proper rc number for msi ' +
> > +                         'version, exiting')
>
> Wondering if this will work for tagged releases that are not RCs. My
> python regexp is a bit rusty, but it seems that out of -for example-
> 0.12.1, the regexp would match 1 as rc_number_match, and add it again to
> the version string. Or if it cannot match an rc number, the else path will
> fail.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joyent/node/pull/14412/files#r27760454>.
>